Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php

```diff
@@ -0,0 +1,683 @@
+                    'removeFromDeskRoundtrip'
+                ) . '&page=manage&replaceSignal=' . $roundtrip_modal->getReplaceSignal()->getId()
+            );
+            $this->addBlockCommand(
+                $this->ctrl->getLinkTarget($this, 'manage'),
+                $this->getRemoveMultipleActionText(),
+                '',
+                $roundtrip_modal
+            );
+        }
+    }
+
+    public function executeCommand(): string
+    {
+        $next_class = $this->ctrl->getNextClass();
+        $cmd = $this->ctrl->getCmd('getHTML');
+
+        switch ($next_class) {
+            case 'ilcommonactiondispatchergui':
+                $gui = ilCommonActionDispatcherGUI::getInstanceFromAjaxCall();
+                if ($gui instanceof ilCommonActionDispatcherGUI) {
+                    $this->ctrl->forwardCommand($gui);
+                }
+                break;
+
+            default:
+                if (method_exists($this, $cmd . 'Object')) {
+                    return $this->{$cmd . 'Object'}();
+                }
+        }
+        return "";
+    }
+
+    public function viewDashboardObject(): void
+    {
+        $this->initAndShow();
+    }
+
+    public function changePDItemSortingObject(): void
+    {
+        $this->viewSettings->storeActorSortingMode(
+            ilUtil::stripSlashes((string) ($this->http->request()->getQueryParams()['sorting'] ?? ''))
+        );
+
+        $this->initAndShow();
+    }
+
+    public function changePDItemPresentationObject(): void
+    {
+        $this->viewSettings->storeActorPresentationMode(
+            ilUtil::stripSlashes((string) ($this->http->request()->getQueryParams()['presentation'] ?? ''))
+        );
+        $this->initAndShow();
+    }
+
+    protected function cancel(): void
+    {
+        $this->ctrl->returnToParent($this);
+    }
+
+    protected function returnToContext(): void
+    {
+        if ($this->http->request()->getQueryParams()['manage'] ?? false) {
+            $this->ctrl->redirect($this, 'manage');
+        }
+        $this->ctrl->redirectByClass('ildashboardgui', 'show');
+    }
+
+    /**
+     * @return array<string, ilBlockDTO[]>
+     */
+    public function getItemGroups(): array
+    {
+        switch ($this->viewSettings->getEffectiveSortingMode()) {
+            case ilPDSelectedItemsBlockConstants::SORT_BY_ALPHABET:
+                $data = $this->getData();
+                $data = array_merge(...array_values($data));
+                $data = $this->sortByTitle($data);
+                return ['' => $data];
+            case ilPDSelectedItemsBlockConstants::SORT_BY_START_DATE:
+                return $this->groupItemsByStartDate();
+            case ilPDSelectedItemsBlockConstants::SORT_BY_TYPE:
+                return $this->groupItemsByType();
+            case ilPDSelectedItemsBlockConstants::SORT_BY_LOCATION:
+            default:
+                return $this->groupItemsByLocation();
+        }
+    }
+
+    #[NoReturn]
```

Comment:
Feel free to use `NoReturn`.
However since that function end with an exit you may better use the [`never`](https://www.php.net/manual/de/language.types.never.php) return type, wich implies NoReturn.

But i encourage you to add #[NoReturn] to (prolly most of) your other functions wich return void since its a heavy impovement for developing within IDEs.

Reply:
Or, if you can, remove the exit like in https://github.com/ILIAS-eLearning/ILIAS/pull/4941#discussion_r1182484724

Reply:
The `NoReturn` / `never` is still valid, as the `$DIC->http()->close()` is effectively an `exit` (but a testable exit).

Reply:
I agree but would not recommend to set `nether` until the returntype of `close()` was changed, too.
Otherwise we might encounter problems with the function chain.

But `NoReturn` should still be possible.